### DISCUSSION TITLE
Guest generation no longer divided by 4 above 7000 guests with NSF

### DISCUSSION THREAD

**Discussion by CSEverett** [OpenRCT2/OpenRCT2#15823] [Feature requests]
This is a pretty simple suggestion. Vanilla and the game currently divide the number of guests generated by 4 when there are more than 7000 guests in the park. With a 9600 sprite limit at the time, this is pretty clearly to help avoid running into it. Now that the NSF no longer has that limit, it follows that guest generation shouldn't be reduced above 7000 guests. Alternately, it could be reduced by a lesser amount (divided by, say, 1.5 or two) instead, to avoid overfilling smaller parks. Or it could be a tickbox option, such as during scenario creation.

**Comment by ZeeMaji** [comment]
I was thinking about this again recently, now that we have the option to have guest goals in scenarios over 7000 (up to 50000) it'd probably make sense change this limit as currently parks with guest goals of over 7000 are only beatable with advertising or weird exploits.

A thought i had is what if this limit was dynamically altered by the guest goal of the current scenario itself? If the guest goal of the current scenario is 7000 or less then the current behavior would be used so old scenarios would remain as is. But if it's more than 7000 then guest generation would be divided by 4 upon hitting the guest goal's guest count + 2000 (same amount of "extra" guests beyond the goal as vanilla)

Eg: A scenario with a guest goal of 12000 would have it's natural guest generation divided by 4 at 14000 guests.

Another option could be having it as a scenario option that could let you set it manually on a per scenario basis but i feel like that might be a bit unneeded and could cause confusion due to how obscure this mechanic is.

Edit: After a bit of discussion in the discord, perhaps having it as a scenario option would be the best option. Basssiiie suggested calling it "harder guest generation after # guests" and it would default to 7000 in old parks.

**Reply by ocalhoun6** [reply]
As far as I know, dividing the guest generation when over 7000 was *only* there to avoid having more guests in the park than the game could handle. It has nothing to do with scenario goals, because if you hit 7000 guests, you were always *well* over the scenario goal anyway (in vanilla).

So the cap should be increased/removed entirely, regardless of scenario goals.

Even in a scenario with a very low guest goal, there's no remaining reason to limit guest generation ... at all.

The only reasons guest generation should ever be limited based on number of guests are:

- It's approaching the (new) sprite limit.
- There are so many guests in the park that hardware limitations will drastically reduce performance, even on very high-tier hardware.

The original limit wasn't put in there because of scenario goals -- so why should it stay for scenario goals? It's not like it's providing any additional challenge for completing the scenario, since if you hit the limit, you're already well above the scenario goal anyway.

**Reply by ZehMatt** [reply]
Something to keep in mind, there might be existing scenarios relying on this.

**Reply by CSEverett** [reply]
Incorporating a change as a tickbox option would be best way to do it.

**Reply by ZeeMaji** [reply]
Yeah that's kind of why i'm leaning more towards a scenario option or otherwise keeping the limit the same for old scenarios, outright removing the limit might change some scenarios as rare as they may be (it'd have to be scenarios with non-guest goals)

**Comment by cheweytoo** [comment]
Heh, I was just about to suggest the same thing: I have a big park with a soft guest cap of over 14000 that still struggles to get more than 7000 guests. Lifting/increasing that limit – now that this is possible – makes a lot of sense.
